### PR TITLE
Affiliate Opt In Feature Documentation

### PR DESCRIPTION
**Pull request by mendelB**
### What does this PR do?

This PR adds documentation for the Affiliate Opt-In Feature

### What are the relevant tickets/cards?

Refs [Pivotal ID #164556255](https://www.pivotaltracker.com/story/show/164556255)
#1370 
#1375 

### Checklist

* [x] Documentation added for new features/changed endpoints.

### CODE REVIEW

File: docs/development/features/affiliate-opt-in.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Affiliate Opt In
+
+The Affiliate Opt In Feature consists of an Opt In Check-box displayed atop the Signup button in the Lede Banner of a campaign.
```
Comment (weerd) [inline docs/development/features/affiliate-opt-in.md]: Add `# Overview` at the top. Been using that in content type docs for the general info on it.

File: docs/development/features/affiliate-opt-in.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	details: "{"campaignContentfulId":"jkl","affiliateOptIn":true}",
+	created_at: "2019-04-20 04:20:00",
+	updated_at: "2019-04-20 04:20:00",
+	deleted_at: null,
+}
+```
+
+This marks this user as having opted in to receiving affiliate messaging, and can be used to help compile a list of such users for the affiliate partner.
+
+## Under the Hood
```
Comment (weerd) [inline docs/development/features/affiliate-opt-in.md]: I used `## Technical Notes` in the content types, but I think I kinda like this title better, haha

Comment (mendelB) [reply docs/development/features/affiliate-opt-in.md]: lolll is this too jovial? do you prefer I switch?

Comment (weerd) [reply docs/development/features/affiliate-opt-in.md]: Haha, nope. I actually think this is better. I'm going to change the other ones to "Under The Hood"

oh, only thing capitalize `The`. I just find it's all more consistent if the rule in titles is to title-case All The Things.

Comment (mendelB) [reply docs/development/features/affiliate-opt-in.md]: haha awesome! Ok will do!

**Pull request merged**